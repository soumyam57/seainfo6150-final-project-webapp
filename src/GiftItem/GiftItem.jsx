import React from 'react';
import PropTypes from "prop-types";
import { useRouteMatch } from "react-router-dom";
import styles from "./GiftItem.module.css";
import GiftItemImage from '../GiftItemImage/GiftItemImage';

const GiftItem = props => {
    return (
      <div className={styles.giftItem}>
        <p  className={styles.text}>{props.giftItem.title}</p>
        <GiftItemImage url = {props.giftItem.img.url} /> 
        <p className={styles.text}>Price: {props.giftItem.price}</p>
      </div>
    )
}

GiftItem.propTypes = {
    giftItem: PropTypes.object.isRequired
};

export default GiftItem;



/*
<giftItem className={styles.giftItem}>
           <div className={styles.wrapper}>
             <p>Price: {props.giftItem.price}</p>
            <GiftItemImage
                url = {props.giftItem.img.url}  
            /> 
            <p>{props.giftItem.title}</p>
          </div>
        </giftItem>*/
