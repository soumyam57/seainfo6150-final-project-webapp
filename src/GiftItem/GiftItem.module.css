.giftItem{
  font-weight: bolder;
  color: gainsboro;
  align-items: center;
  justify-items: center;
  text-align: center;
  box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
  border: 0px solid #666;
  border-radius: 5%;
}


.giftItem:hover {
  border: 5px solid maroon;
  background-color:rgb(99, 54, 54);
  text-decoration-color: maroon;
}


.text {
    padding: 4%;
    color: gainsboro;
    font-weight: bolder;
    text-decoration-line: none;
    text-align: center;
}

  
  @media screen and (min-width: 525px) {
    .giftItem {
      padding: 1%;
    }
    .title {
      font-size: 1.5em;
    }
  }
  
  @media screen and (min-width: 800px) {
    .giftItem {
      padding: 1%;
    }
    .title {
      font-size: 1em;
    }
  }
  
  @media screen and (min-width: 1150px) {
    .giftItem {
      padding: 1%;
    }
    .title {
      font-size: 0.8em;
    }
  }
  