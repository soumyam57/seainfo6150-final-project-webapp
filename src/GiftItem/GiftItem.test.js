import React from "react";
import { render } from "@testing-library/react";
import GiftItem from "./GiftItem.jsx";

const giftItem = {
    title: "Couple Rings",
    img:
      {
         "url":"https://images-na.ssl-images-amazon.com/images/I/51mfwq0NDEL._AC_UX522_.jpg",
         "height":400,
         "width":500
      },
    price: "$1000"
}

it("renders the home component correctly", () => {
  const { container } = render(
    <GiftItem
        giftItem={giftItem}
    />
    );
  expect(container).toMatchSnapshot();
});
