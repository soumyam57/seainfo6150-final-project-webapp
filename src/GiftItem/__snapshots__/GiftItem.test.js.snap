// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div
    class="giftItem"
  >
    <p
      class="text"
    >
      Couple Rings
    </p>
    <div>
      <img
        alt=""
        class="image"
        src="https://images-na.ssl-images-amazon.com/images/I/51mfwq0NDEL._AC_UX522_.jpg"
      />
    </div>
    <p
      class="text"
    >
      Price: 
      $1000
    </p>
  </div>
</div>
`;
